pkg/apply/prune/prune_test.go

```diff
@@ -159,18 +159,33 @@ func TestUnionPastInventory(t *testing.T) {
 
 	for name, tc := range tests {
 		t.Run(name, func(t *testing.T) {
-			actual, err := unionPastInventory(tc.groupingInfos)
-			expected := NewInventory(tc.expected)
+			actual, err := unionPastObjs(tc.groupingInfos)
 			if err != nil {
 				t.Errorf("Unexpected error received: %s\n", err)
 			}
-			if !expected.Equals(actual) {
-				t.Errorf("Expected inventory (%s), got (%s)\n", expected, actual)
+			if len(tc.expected) != len(actual) {
+				t.Fatalf("Expected (%d) objects, got (%d)\n", len(tc.expected), len(actual))
+			}
+			for _, expectedObj := range tc.expected {
+				if !objInArray(expectedObj, actual) {
+					t.Fatalf("Expected object (%s), but not found\n", expectedObj)
+				}
 			}
 		})
 	}
 }
 
+// objInArray is a helper function that returns true if passed obj
+// is in array of objects; false otherwise.
+func objInArray(obj *object.ObjMetadata, arr []*object.ObjMetadata) bool {
+	for _, a := range arr {
+		if a.String() == obj.String() {
```

Comment:
I think we should do a proper `Equals` function for this. Or if we can avoid using pointers here, I think we could just use `a == obj`.

Reply:
I think I've addressed your concern. I've avoided pointers here, and used `a == obj`. Let me know if this works.